### ISSUE TITLE
Subcommands doesn't work on windows

### ISSUE THREAD

**Issue by julianduque** [leo/args#88]
The subcommand binary isn't found on windows environments

```
C:\> tool config get option
C:\Users\jduque\AppData\Roaming\npm\node_modules\tool\node_modules\args\index.js:342
throw err
^

Error: spawn tool.js-config ENOENT
```

**Comment by ntwcklng** [comment]
Hey, did you try this: #46?

**Comment by julianduque** [comment]
@ntwcklng yes, I have the binary scripts in `package.json` but it fails only on Windows, Linux/mac works good

**Comment by ntwcklng** [comment]
Unfortunately i don't own a windows machine, can anyone else help & debug?

**Comment by TooTallNate** [comment]
Could be worth setting up AppVeyor tests for a tool like this.

**Comment by ntwcklng** [comment]
@julianduque can you please test if this works?
Replace this https://github.com/leo/args/blob/master/lib/utils.js#L281-L283
with this:
```js
    if (process.platform === 'win32') {
      // Run binary of sub command on windows
      args.unshift(full)
      this.child = spawn(process.execPath, args, {
        stdio: 'inherit'
      });
    } else {
      // Run binary of sub command
      this.child = spawn(full, args, {
        stdio: 'inherit'
      });
    }

```

**Comment by ntwcklng** [comment]
I now have access to a windows machine and came up with this fix: https://github.com/leo/args/commit/bf222031fbc440473483c0bf3bf90497a2b1196f
But it feels a bit hacky - is there another way to run the binary?

**Comment by ntwcklng** [comment]
cc @leo what do you think about this: bf22203

**Comment by leo** [comment]
Looks good to me as a start, @ntwcklng! 😊

**Comment by Gioyik** [comment]
@ntwcklng I tested your solution but still not luck. This is what I get now:

```
C:\> mytool config get option
module.js:471
  throw err
  ^

Error: Cannot find module 'C:\Users\gioyik\Documents\MyProject\mytool-config.js'
```

The `package.json` of `mytool` I have:

```js
  "bin": {
    "mytool": "./bin/mytool.js",
    "mytool-config": "./bin/mytool-config.js"
  },
```

Seems it's looking for `mytool-config.js` file inside the project/folder I run `mytool`.  Hope this helps to understand what is happening.

**Comment by ntwcklng** [comment]
@Gioyik Thanks for your help! I've pushed another commit to #105 - can you check it out?

**Comment by Gioyik** [comment]
@ntwcklng is working now. I think the PR is ready to be merge now that is working as expected on Windows.

**Comment by julianduque** [comment]
Thanks @Gioyik for confirm and @ntwcklng for your work on this <3 - now let's wait for @leo review  🙇

**Comment by ntwcklng** [comment]
@Gioyik, @julianduque released in `3.0.5`. Many thanks!

**Comment by Gioyik** [comment]
@ntwcklng thank you, we appreciate the really fast answer from you and Leo.
